panels/UserPanel.php

```diff
@@ -116,4 +108,36 @@ protected function dataToString($data)
 
         return VarDumper::export($data);
     }
+
+    /**
+     * Returns an array containing information about the logged-in user.
+     *
+     * The array should contain two items:
+     * - the model that should be set on [[\yii\widgets\DetailView::model]]
+     * - the array that should be set on [[\yii\widgets\DetailView::attributes]]
+     *
+     * @param IdentityInterface $identity
+     * @return array
+     */
+    protected function identityData(IdentityInterface $identity)
+    {
+        if ($identity instanceof Model) {
+            $data = $identity->getAttributes();
+            $attributes = [];
+
+            foreach ($data as $attribute => &$value) {
+                $attributes[] = [
+                    'attribute' => $attribute,
+                    'label' => $identity->getAttributeLabel($attribute)
+                ];
+            }
+            unset($value);
+        } else {
+            $data = get_object_vars($identity);
+            // Let the DetailView widget figure the labels out
+            $attributes = null;
+        }
+
+        return [$data, $attributes];
```

Comment:
which part is important for you for extension in subclasses? (if I interpretate your introductionary comment correctly)